File: Algorithms/CS/SitoEratostenesa.cs
namespace SitoErastotenesa
{
    class Erastotenes
    {
        static void Main(String[] args)
        {
            Console.WriteLine("Sito Erastotenesa: ");
            Erastotenes fun = new Erastotenes();
            fun.Oblicz(200);
        }

        public void Oblicz(int x)
        {
            int[] tab = new int[x];

            for (int i = 0; i < x; i++)
                tab[i] = 0;

            for(int i=2;i<= (int)Math.Sqrt(x);i++)
            {
                int index = i * 2;
                while(index < x)
                {
                    tab[index] = 1;
                    index += i;
                }
            }

            for (int i = 2; i < x; i++)
                if (tab[i] == 0)
                    Console.Write(i + ", ");
        }
    }
}

File: Algorithms/Kotlin/SitoEratostenesa.kt
fun SitoErastotenesa(x: Int)
{
    println(x);
    var tab = IntArray(x);

    var i = 0;
    while(i < x)
    {
        tab[i] = 0
        i++
    }

    i = 2

    while(i <= (sqrt(x.toDouble())).toInt())
    {
        var index: Int = i*2;
        while(index < x)
        {
            tab[index] = 1
            index += i;
        }
        i++
    }



    i = 2;
    while(i < x)
    {
        if(tab[i] == 0)
            print("$i, ")
        i++
    }

}
